### ISSUE TITLE
0.20.4 - hacs no longer sees installed components

### ISSUE THREAD

**Issue by Dinth** [hacs/integration#914]
## Version of HACS
0.20.4 

## Describe the bug
HACS no longer sees custom components and lovelace plugins installed before. 
![Screenshot from 2020-01-02 22-31-51](https://user-images.githubusercontent.com/313674/71697405-d0925880-2daf-11ea-8cdb-005470be67ae.png)

Additionally every few restarts of HA it shows onboarding help. 

## Debug log

```text
2020-01-02 22:26:50 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for healthchecksio which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2020-01-02 22:26:50 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2020-01-02 22:26:57 DEBUG (MainThread) [hacs] Configuration type: flow
2020-01-02 22:26:57 INFO (MainThread) [hacs]
-------------------------------------------------------------------
HACS (Home Assistant Community Store)
  
Version: 0.20.4
This is a custom integration
If you have any issues with this you need to open an issue here:
https://github.com/hacs/integration/issues
-------------------------------------------------------------------

2020-01-02 22:26:57 DEBUG (MainThread) [hacs.repository.integration.hacs.integration] Checking repository.
2020-01-02 22:27:01 ERROR (SyncWorker_18) [homeassistant.components.xbox_live.sensor] Can't setup XboxAPI connection. Check your account or api key on xboxapi.com. Code: 401 Description: A fresh login is required to gain a new token from Microsoft
2020-01-02 22:27:04 INFO (MainThread) [hacs.repository.integration.hacs.integration] Registration complete
2020-01-02 22:27:04 INFO (MainThread) [hacs.data] Restore started
2020-01-02 22:27:04 INFO (MainThread) [hacs.data] Restore done
2020-01-02 22:27:08 ERROR (SyncWorker_4) [homeassistant.components.qbittorrent.sensor] Connection failed
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 351, in async_enable_automation
    self.async_trigger
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 482, in _async_process_trigger
    remove = await platform.async_attach_trigger(hass, conf, action, info)
  File "/usr/src/homeassistant/homeassistant/components/automation/time_pattern.py", line 53, in async_attach_trigger
    hass, time_automation_listener, hour=hours, minute=minutes, second=seconds
  File "/usr/src/homeassistant/homeassistant/helpers/event.py", line 471, in async_track_time_change
    return async_track_utc_time_change(hass, action, hour, minute, second, local=True)
  File "/usr/src/homeassistant/homeassistant/helpers/event.py", line 418, in async_track_utc_time_change
    matching_hours = dt_util.parse_time_expression(hour, 0, 23)
  File "/usr/src/homeassistant/homeassistant/util/dt.py", line 230, in parse_time_expression
    res = list(sorted(int(x) for x in parameter))
  File "/usr/src/homeassistant/homeassistant/util/dt.py", line 230, in <genexpr>
    res = list(sorted(int(x) for x in parameter))
ValueError: invalid literal for int() with base 10: ','
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.meteoalarm_effective fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 281, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 459, in async_device_update
    await self.async_update()
  File "/usr/src/homeassistant/homeassistant/components/template/sensor.py", line 224, in async_update
    self._state = self._template.async_render()
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 222, in async_render
    return compiled.render(kwargs).strip()
  File "/usr/local/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/usr/local/lib/python3.7/site-packages/jinja2/sandbox.py", line 440, in call
    return __context.call(__obj, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 814, in strptime
    return datetime.strptime(string, fmt)
TypeError: strptime() argument 1 must be str, not Undefined
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.components.template.sensor] Could not render template Urgency: UndefinedError: 'mappingproxy object' has no attribute 'urgency'
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.meteoalarm_onset fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 281, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 459, in async_device_update
    await self.async_update()
  File "/usr/src/homeassistant/homeassistant/components/template/sensor.py", line 224, in async_update
    self._state = self._template.async_render()
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 222, in async_render
    return compiled.render(kwargs).strip()
  File "/usr/local/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/usr/local/lib/python3.7/site-packages/jinja2/sandbox.py", line 440, in call
    return __context.call(__obj, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 814, in strptime
    return datetime.strptime(string, fmt)
TypeError: strptime() argument 1 must be str, not Undefined
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.components.template.sensor] Could not render template Severity: UndefinedError: 'mappingproxy object' has no attribute 'severity'
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.meteoalarm_expires fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 281, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 459, in async_device_update
    await self.async_update()
  File "/usr/src/homeassistant/homeassistant/components/template/sensor.py", line 224, in async_update
    self._state = self._template.async_render()
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 222, in async_render
    return compiled.render(kwargs).strip()
  File "/usr/local/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/usr/local/lib/python3.7/site-packages/jinja2/sandbox.py", line 440, in call
    return __context.call(__obj, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 814, in strptime
    return datetime.strptime(string, fmt)
TypeError: strptime() argument 1 must be str, not Undefined
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.meteoalarm_colour fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 281, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 459, in async_device_update
    await self.async_update()
  File "/usr/src/homeassistant/homeassistant/components/template/sensor.py", line 224, in async_update
    self._state = self._template.async_render()
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 222, in async_render
    return compiled.render(kwargs).strip()
  File "/usr/local/lib/python3.7/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 864, in regex_findall_index
    return re.findall(find, value, flags)[index]
IndexError: list index out of range
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.components.template.sensor] Could not render template Certainty: UndefinedError: 'mappingproxy object' has no attribute 'certainty'
2020-01-02 22:27:08 ERROR (MainThread) [homeassistant.components.automation] Error while executing automation automation.ha_start_notification. Error rendering template for call_service at pos 1: UndefinedError: 'sensor' is undefined
2020-01-02 22:27:10 DEBUG (MainThread) [hacs] 4823
2020-01-02 22:27:10 DEBUG (MainThread) [hacs] 2020-01-02 23:19:21
2020-01-02 22:27:10 INFO (MainThread) [hacs] Loading known repositories
2020-01-02 22:27:16 INFO (MainThread) [hacs.factory] Prosessing 1 tasks
2020-01-02 22:27:16 DEBUG (MainThread) [hacs.repository.appdaemon.benleb.ad-ench] Checking repository.
2020-01-02 22:27:18 DEBUG (MainThread) [hacs] 4811
2020-01-02 22:27:18 DEBUG (MainThread) [hacs] 2020-01-02 23:19:21
2020-01-02 22:27:18 ERROR (MainThread) [hacs] Validation for benleb/ad-ench failed with GitHub returned 404 for https://api.github.com/repos/benleb/ad-ench/contents/apps.
2020-01-02 22:27:23 INFO (MainThread) [hacs.factory] Task prosessing of 1 tasks completed in 7 seconds
2020-01-02 22:27:23 DEBUG (MainThread) [hacs] Starting recuring background task for installed repositories
2020-01-02 22:27:23 DEBUG (MainThread) [hacs] 4811
2020-01-02 22:27:23 DEBUG (MainThread) [hacs] 2020-01-02 23:19:21
2020-01-02 22:27:23 INFO (MainThread) [hacs.factory] Prosessing 1 tasks
2020-01-02 22:27:23 DEBUG (MainThread) [hacs.repository.integration.hacs.integration] Getting repository information
2020-01-02 22:27:38 INFO (MainThread) [hacs.factory] Task prosessing of 1 tasks completed in 16 seconds
2020-01-02 22:27:39 DEBUG (MainThread) [hacs.data] Saving data
2020-01-02 22:27:39 DEBUG (MainThread) [hacs] Recuring background task for installed repositories done
2020-01-02 22:27:39 DEBUG (MainThread) [hacs.data] Saving data
2020-01-02 22:27:46 ERROR (SyncWorker_21) [homeassistant.components.qbittorrent.sensor] Connection failed
```

**Comment by ludeeus** [comment]
And what did you do before this happened?
Upgraded from an old version?

**Comment by Dinth** [comment]
As far as i remember i have just restarted HA. After the restart when i clicked Community menu HACS showed me onboarding info. After clicking through it, all my installed (through HACS) integrations and plugins are no longer there.

**Comment by ludeeus** [comment]
Then you have upgraded HACS (you are now on the latest published version)
If you have a backup, you can restore to get them back, and carefully upgrading from the version you have (read all the release notes between the version in your backup and the newest, you may need to update to 0.16.3 first)
Or just reinstalling them.

**Comment by Dinth** [comment]
Hi Ludeus. Yeah, ive been updating HACS recently (but definetely not immediately before this issue to occur). Also ive been updating from 0.20.x as im trying to keep all my extensions up to date.

**Comment by DamianosS** [comment]
Having the same problem. Currently I'm on 0.20.4. I can go back only to 0.20.0.
Would I have to uninstall and reinstall again?

**Comment by CervDotBe** [comment]
How can you update to 0.16.3 first?

**Comment by ludeeus** [comment]
> Also ive been updating from 0.20.x as im trying to keep all my extensions up to date.

If that is the case then one of these are true:
- You have not restarted Home Assistant between updates (so it still used an old version)
- At some point, there was an error during startup.

If you really want it back, restore from a backup, HACS stores that info in `.storage/hacs.repositories`
Or just reinstall the elements you had.

Same goes for you @DamianosS 

> How can you update to 0.16.3 first?

Find that release and download the .zip file.

**Comment by DamianosS** [comment]
What I did that worked for me:
- uninstalled HACS (via the UI)
- restarted HA
- reinstalled it (via the UI)
- restarted HA
- reinstalled all plugins (via the UI)

I'm on 0.20.7 now, all green.

BTW, thanks a lot for the improved UI. Much more comfortable and informative!

**Comment by colonelmattyman** [comment]
> > How can you update to 0.16.3 first?
> 
> Find that release and download the .zip file.

Most people would have installed this via the Community store and probably don't know how to install something manually.  The community store currently has a giant warning.
"DO NOT UPGRADE TO THE LATEST VERSION!
First upgrade to version 0.16.3, then upgrade to the latest version."
The versions that can be downloaded in the "Available versions" do not contain this version number.  Can I suggest adding 0.16.3 in as an option to make it easy for people to fix?

**Comment by ludeeus** [comment]
No one installed HACS through HACS, that would not be possible.

**Comment by colonelmattyman** [comment]
> No one installed HACS through HACS, that would not be possible.

Haha.  You're right.  That was a pretty dumb comment.

Still ...
The versions that can be downloaded in the "Available versions" do not contain the 0.16.3 version number. Can I suggest adding 0.16.3 in as an option to make it easy for people to fix?

**Comment by ludeeus** [comment]
Just add it yourself https://hacs.xyz/docs/configuration/legacy#options-map
